.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,58 @@
+name: continuous-integration
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+  release:
+    types: [published]
+
+jobs:
+  continuous-integration:
+    name: continuous-integration
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: install dependencies
+        run: npm install
+
+      - name: lint site
+        run: npm run lint
+
+      - name: build site
+        run: npm run build
+
+  configure:
+    name: configure deployment target environment
+    runs-on: ubuntu-20.04
+    outputs:
+      environment: ${{ steps.environment-identifier.outputs.environment }}
+    steps:
+      - id: environment-identifier
+        run: |
+          if [[ "${{ env.GITHUB_REF_NAME }}" == "master" ]];
```

Comment:
The default branch is not `master` here. Maybe we should use either the "default branch dynamically collected" or `main`